docs/design/selector-generation.md

```diff
@@ -110,18 +110,42 @@ No changes
 ## Kubectl
 
 No required changes.
-Suggest moving SELECTOR to wide output of `kubectl get jobs` since users don't write the selector.
+Suggest moving SELECTOR to wide output of `kubectl get jobs` since users do not write the selector.
 
 ## Docs
 
 Remove examples that use selector and remove labels from pod templates.
 Recommend `kubectl get jobs -l job-name=name` as the way to find pods of a job.
 
-# Cross Version Compat
+# Conversion
 
-`v1beta1` will not have a `job.spec.manualSelector` and will not provide a default selector.
+The following applies to Job, as well as to other types that adopt this pattern.
+
+Type `extensions/v1beta1` will not have a `job.spec.manualSelector`.  Both the internal type and the `batch/v1` type will have
```

Comment:
If this field is present in a preexisting extensions/v1beta1 object, it must be set to explicit true, since the selectors are manually generated, and would thus fail the validation if not set this way.

People may take an existing job by GETing it, and then edit it to convert it to batch/v1 and then POST it again.  I really do not want them to pick up the "manualSelector=true" value and propagate that.  This is less of an issue for Job since they probably won't be converted while "still alive".  However, it may be an issue for ReplicaSet, so I wanted to work out a system that could be applied there (though, I'm still not sure how it works with ReplicaSet if you want to pods to stay alive during the upgrade: I think you end up with manualSelector in that case due to adoption).